### PR TITLE
Change count_variant_alleles to not reconstruct DataArray.

### LINKED ISSUE #305: Change count_variant_alleles to not reconstruct DataArray
I noticed that `count_variant_alleles` evaluated immediately when running it on a client to a dask cluster.  That may be because numba isn't installed on the workers.  If that was it, we should document this in a "troubleshooting" section of the docs.

### PR DESCRIPTION
**Pull request by tomwhite**
Test that all allele counting functions are lazy.

Note that `test_count_variant_alleles__chunked` fails without the change in `count_variant_alleles`. 

Fixes #305.

### CODE REVIEW

File: sgkit/stats/popgen.py [hunk 1]
```diff
@@ -406,7 +406,7 @@ def Tajimas_D(
     S = ((ac > 0).sum(axis=1) > 1).sum()
 
     # assume number of chromosomes sampled is constant for all variants
-    n = ac.sum(axis=1).max()
+    n = ac.sum(axis=1).max().compute()
```
Comment (tomwhite) [inline sgkit/stats/popgen.py:409]: I'm not sure if there's a way to make `Tajimas_D`, so this `compute` is needed for the time being at least. This was effectively being hidden by `variant_allele_count` not being lazy.

**Review by jeromekelleher — APPROVED** [review]
LGTM

**Comment by jeromekelleher** [comment]
Doesn't look like anything controversial, feel free to add the merge label if you're happy with merging.

File: sgkit/stats/popgen.py (hunk 1 above)
Comment (ravwojdyla) [reply sgkit/stats/popgen.py:409]: @tomwhite we should probably open an issue to remove this compute, and make it lazy again?

Comment (ravwojdyla) [reply sgkit/stats/popgen.py:409]: FYI https://github.com/pystatgen/sgkit/issues/354

**Pull request merged**